src/main/java/com/linecorp/armeria/common/logging/MessageLogBuilder.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.logging;
+
+public interface MessageLogBuilder {
+    void increaseContentLength(long deltaBytes);
+    void contentLength(long contentLength);
+    void attach(Object attachment);
+    void end(Throwable cause);
```

Comment:
by method signature, I feel there is no way to end build without exception. how about adding some syntactic-sugar method `end()` which just calls `end(null)`? maybe it can be implemented using `default`.